tools/ubuntu-setup.sh

```diff
@@ -0,0 +1,272 @@
+    echo "nameserver 8.8.8.8" | sudo tee -a $IMG_BLD_DIR/etc/resolve.conf
+    echo "syzkaller" | sudo tee $IMG_BLD_DIR/etc/hostname
+    ssh-keygen -f "$IMG_BLD_DIR/$IMG_RELEASE.id_rsa" -t rsa -N ''
+    sudo mkdir -p "$IMG_BLD_DIR/root/.ssh/"
+    cat "$IMG_BLD_DIR/$IMG_RELEASE.id_rsa.pub" | sudo tee $IMG_BLD_DIR/root/.ssh/authorized_keys
+
+    sudo chroot $IMG_BLD_DIR /bin/bash -c "apt-get update; apt-get install -y curl tar time strace gcc make sysbench git vim screen usbutils"
+    sudo chroot $IMG_BLD_DIR /bin/bash -c "mkdir -p ~; cd ~/; wget https://github.com/kernelslacker/trinity/archive/v1.5.tar.gz -O trinity-1.5.tar.gz; tar -xf trinity-1.5.tar.gz"
+    sudo chroot $IMG_BLD_DIR /bin/bash -c "cd ~/trinity-1.5 ; ./configure.sh ; make -j4 ; make install"
+
+    # Build a disk image
+    dd if=/dev/zero of="$IMG_FILE" bs=1M seek=2047 count=1
+    sudo mkfs.ext4 -F "$IMG_FILE"
+    sudo mkdir -p /mnt/$IMG_RELEASE
+    sudo mount -o loop "$IMG_FILE" /mnt/$IMG_RELEASE
+    sudo cp -a $IMG_BLD_DIR/. /mnt/$IMG_RELEASE/.
+    sudo umount /mnt/$IMG_RELEASE
+}
+
+function build_img {
+    echo "[*] build stretch img"
+    if [[ ! -f "$IMG_FILE" ]]; then
+	echo "[*] $IMG_FILE is building"
+	create_image
+    else
+	echo "[*] $IMG_FILE FOUND"
+    fi
+}
+
+function boot_img {
+    sudo qemu-system-x86_64 -kernel "$KBLD_DIR/arch/x86/boot/bzImage" \
+    -append "console=ttyS0 root=/dev/sda debug earlyprintk=serial slub_debug=QUZ" \
+    -hda "$IMG_FILE" \
+    -net user,hostfwd=tcp::10021-:22 \
+    -net nic -enable-kvm -nographic -m 2G -smp 2 \
+    -pidfile "$INIT_DIR/vm.pid" 2>&1 | tee "$INIT_DIR/vm.log"
+}
+
+function print_ssh_cmd {
+    echo "[*] ssh -i \"$IMG_BLD_DIR/$IMG_RELEASE.id_rsa\" -p 10021 -o \"StrictHostKeyChecking no\" root@localhost"
+}
+
+function config_syzkaller {
+    if [[ ! -f "$INIT_DIR/$GOLANG_FILE" ]]; then
+	echo "[*] DOWNLOAD $GOLANG_FILE"
+	wget -O "$INIT_DIR/$GOLANG_FILE" -c "$GOLANG_URL"
+    else
+	echo "[+] $GOLANG_FILE FOUND"
+    fi
+
+    echo "$INIT_DIR/$GOLANG_FILE"
+    cd "$INIT_DIR" && tar xvf "$INIT_DIR/$GOLANG_FILE" && mkdir "$GOPATH"
+    go get -u -d github.com/google/syzkaller/...
+    cd "$GOPATH/src/github.com/google/syzkaller/" && make
+    # mkdir "$INIT_DIR/syzkaller_workdir"
+
+    cat > "$INIT_DIR/syzkaller.cfg" << EOF
+{
+	"target": "linux/amd64",
+	"http": "127.0.0.1:56741",
+	"workdir": "$INIT_DIR/syzkaller_workdir",
+        "kernel_obj": "$KBLD_DIR",
+	"image": "$IMG_FILE",
+	"sshkey": "$IMG_BLD_DIR/$IMG_RELEASE.id_rsa",
+	"syzkaller": "$GOPATH/src/github.com/google/syzkaller",
+	"procs": 8,
+	"type": "qemu",
+	"vm": {
+		"count": 4,
+		"kernel": "$KBLD_DIR/arch/x86/boot/bzImage",
+		"cpu": 2,
+		"mem": 2048
+	}
+}
+EOF
+
+    sudo "$GOPATH/src/github.com/google/syzkaller/bin/syz-manager" -config "$INIT_DIR/syzkaller.cfg" -debug
+}
+
+install_dependencies
+check_gcc_coverage
+patch_linux_kernel
+build_linux_kernel
+build_img
+print_ssh_cmd
+# boot_img
+config_syzkaller
+
+# references
```

Comment:
I think it's better to put references at the top, because not everybody will read down to here.